Announce opam 2.2.0~beta1 (#97)
ocaml/platform-blog: opam-2-2-0-beta1.md

* Fix opam env containing carriage return on Cygwin

## Miscellaneous changes

* Remove stray comments from pwsh and cmd env
* Add ./configure --enable-static to compile the opam binary statically on Linux
* Fix debug logs showing up regardless of verbosity on macOS 12.7.1 / 13.6.3 / 14.2 and FreeBSD
* Upgrade to, and require mccs >= 1.1+17
* Fix `opam tree --dev`
* Fix opam tree --no-switch. Instead of emptying the current switch from its installed packages, it load a fresh virtual switch

Various other improvements were made and bugs were fixed.
API changes are denoted in the release note linked above.
This release also includes a handful of PRs improving the documentation and more than a dozen PRs improving and extending the tests.

## Windows Support

The main opam-repository Windows compliance is still a work in progress, we
recommend to use existing [compatible
repository](https://github.com/ocaml-opam/opam-repository-mingw) (originally
from [@fdopen](https://github.com/fdopen)) and [32/64 bit mingw-w64
packages](https://github.com/dra27/opam-repository/tree/windows-5.0) (by
[@dra27](https://github.com/dra27)).

### How to Test opam on Windows

> If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-preview-for-windows.exe).
> 
> Otherwise you can compile opam yourself using the following steps:

Author: While I’m figuring out the Windows executable
```suggestion
<!--
If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-preview-for-windows.exe).

Otherwise you can compile opam yourself using the following steps:
---!>
You can compile opam yourself using the following steps:
```

* Fix `opam env` containing carriage return on Cygwin - `eval $(opam env)` now works from a Cygwin bash terminal

## Miscellaneous changes

* Remove stray comments from `pwsh` and `cmd` opam env output
* Add `./configure --enable-static` to compile the opam binary statically on Linux
* Fix debug logs showing up regardless of verbosity on macOS 12.7.1 / 13.6.3 / 14.2 and FreeBSD
* Upgrade to, and require mccs >= 1.1+17
* Fix `opam tree` options `--dev` and `--no-switch`

Various other improvements were made and bugs were fixed.
API changes are denoted in the release note linked above.
This release also includes PRs improving the documentation and improving and extending the tests.

## Windows Support

The main opam-repository Windows compliance is still a work in progress, we
recommend to use existing [compatible
repository](https://github.com/ocaml-opam/opam-repository-mingw) (originally
from [@fdopen](https://github.com/fdopen)) and [32/64 bit mingw-w64
packages](https://github.com/dra27/opam-repository/tree/windows-5.0) (by
[@dra27](https://github.com/dra27)).

### How to Test opam on Windows

> If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-preview-for-windows.exe).

Author: ```suggestion
If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-x86_64-preview-for-windows.exe).
```

> If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-x86_64-preview-for-windows.exe).

Reviewer: ```suggestion
If you're feeling adventurous, you can use the experimental pre-built binary for Windows available [here](https://github.com/ocaml/opam/releases/download/2.2.0-beta1/opam-2.2.0-beta1-x86_64-preview-for-windows.exe).
```

title: "opam 2.2.0 beta1 release"
authors: [
  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
  "Kate Deplaix - Tarides" {"mailto:kit-ty-kate(à)outlook.com"}
  "David Allsopp - Tarides" {"mailto:david(à)tarides.com"}
]
date: "2024-01-18"
--BODY--

<!--
_Feedback on this post is welcomed on [Discuss](https://discuss.ocaml.org/t/ann-opam-2-2-0-beta1/?????)!_
---!>

We are happy to announce the first beta release of opam 2.2.0. It contains
some fixes and a new feature for Windows.  You can view the full list in the
[release note](https://github.com/ocaml/opam/releases/tag/2.2.0-beta1).

This version is a beta, we invite users to test it to spot previously
unnoticed bugs as we head towards the stable release.

## Major change: Check and advertise to use Git for Windows

> Opam 2.2 is based on Cygwin install (previously installed, or internally to
> opam). Cygwin Git is known to not have a common behaviour: different set of
> trusted certificates authorities, its own Credential Manager, different Git
> configuration, etc.  We recommend then to install and use [Git for
> Windows](https://gitforwindows.org) or via `winget`.
> 
> At init stage, opam checks available Git(s), and ask users to choose one of the
> already ones, or to rerun `opam init` after installing another Git.

Reviewer 2: ```suggestion
Opam 2.2 is based on a Cygwin installation (previously installed, or managed
internally by opam). Cygwin's Git has three known usability issues when used
outside a Cygwin environment: it uses a different set of trusted certificate
authorities, has its own Credential Manager and potentially uses a separate Git
configuration.  We therefore recommend using [Git for
Windows](https://gitforwindows.org) either installed manually or via `winget`.

At `opam init`, opam checks for available Git(s), and asks the user to choose
one of the available, or to rerun `opam init` after installing another Git.
```

## Other noteworthy changes

* When compiling opam on Windows with MinGW, the resulting opam binary now contains `libstdc++` instead of requiring the DLL to be distributed alongside it or present in the environment

> * Fix opam env containing carriage return on Cygwin

Reviewer 2: ```suggestion
* Fix `opam env` containing carriage return on Cygwin - `$(eval opam env)` now works from a Cygwin bash terminal
```

Author: wouldn’t it be `eval $(opam env)` instead?

Author: ```suggestion
* Fix `opam env` containing carriage return on Cygwin - `eval $(opam env)` now works from a Cygwin bash terminal
```
